### DISCUSSION TITLE
Live update pyplot

### DISCUSSION THREAD

**Discussion by nghia-vo** [zauberzeug/nicegui#405] [Q&A] [answered]
There's push method in https://nicegui.io/reference#line_plot. However, I would like to do the same with https://nicegui.io/reference#pyplot_context to do more than just 1d plotting. How to do that with pyplot? Btw, the example is out-of-date. ui.pyplot has been replaced by ui.plot.

**Comment by rodja** [comment, marked as answer]
`ui.line_plot` is an handy inherit of `ui.pyplot`. I suggest you copy the code from https://github.com/zauberzeug/nicegui/blob/main/nicegui/elements/line_plot.py into your own project. Then you can modify it to your needs.

**Comment by rodja** [comment]
> Btw, the example is out-of-date. ui.pyplot has been replaced by ui.plot.

I think it's the other way around. Since the [introduction of Plotly-Support](https://nicegui.io/reference#plotly_element) we deprecated `ui.plot`. Instead of one generic name `ui.plot` we now use `ui.pyplot` and `ui.plotly`.
